sonic-gnmi-standalone/internal/diskspace/diskspace.go

```diff
@@ -0,0 +1,176 @@
+	"syscall"
+
+	"github.com/golang/glog"
+)
+
+// GetDiskTotalSpaceMB returns the total disk space in MB for the given path.
+func GetDiskTotalSpaceMB(path string) (int64, error) {
+	var stat syscall.Statfs_t
+	if err := syscall.Statfs(path, &stat); err != nil {
+		return 0, fmt.Errorf("failed to get filesystem stats for %s: %w", path, err)
+	}
+
+	// Calculate total space in MB
+	blockSize := int64(stat.Bsize)
+	totalBytes := int64(stat.Blocks) * blockSize
+	totalMB := totalBytes / (1024 * 1024)
+
+	glog.V(2).Infof("Total space for %s: %d MB", path, totalMB)
+	return totalMB, nil
+}
+
+// GetDiskFreeSpaceMB returns the available disk space in MB for the given path.
+func GetDiskFreeSpaceMB(path string) (int64, error) {
+	var stat syscall.Statfs_t
+	if err := syscall.Statfs(path, &stat); err != nil {
+		return 0, fmt.Errorf("failed to get filesystem stats for %s: %w", path, err)
+	}
+
+	// Calculate available space in MB (using Bavail which is available to non-root users)
+	blockSize := int64(stat.Bsize)
+	availBytes := int64(stat.Bavail) * blockSize
+	availMB := availBytes / (1024 * 1024)
+
+	glog.V(2).Infof("Available space for %s: %d MB", path, availMB)
+	return availMB, nil
+}
+
+// CleanupResult represents the result of a cleanup operation.
+type CleanupResult struct {
+	Path         string
+	FilesRemoved int
+	SpaceFreedMB int64
+	Errors       []error
+}
+
+// CleanupOptions configures cleanup behavior.
+type CleanupOptions struct {
+	DryRun bool // If true, only report what would be cleaned without actually deleting
+}
+
+// CleanupCoreFiles removes core dump files from /var/core/*.
+func CleanupCoreFiles(opts CleanupOptions) (*CleanupResult, error) {
+	return cleanupDirectory("/var/core", opts)
+}
+
+// CleanupDumpFiles removes dump files from /var/dump/*.
+func CleanupDumpFiles(opts CleanupOptions) (*CleanupResult, error) {
+	return cleanupDirectory("/var/dump", opts)
+}
+
+// cleanupDirectory removes all files and subdirectories from the specified directory.
+func cleanupDirectory(dirPath string, opts CleanupOptions) (*CleanupResult, error) {
+	result := &CleanupResult{
+		Path:   dirPath,
+		Errors: make([]error, 0),
+	}
+
+	// Check if directory exists
+	if _, err := os.Stat(dirPath); os.IsNotExist(err) {
+		glog.V(2).Infof("Directory %s does not exist, skipping cleanup", dirPath)
+		return result, nil
+	}
+
+	// Get all items in the directory
+	pattern := filepath.Join(dirPath, "*")
+	matches, err := filepath.Glob(pattern)
+	if err != nil {
+		return result, fmt.Errorf("failed to list files in %s: %w", dirPath, err)
+	}
+
+	glog.V(2).Infof("Found %d items to clean in %s (dry-run: %v)", len(matches), dirPath, opts.DryRun)
+
+	// Calculate space and remove files
+	for _, match := range matches {
+		size, err := calculateDirectorySize(match)
+		if err != nil {
+			result.Errors = append(result.Errors, fmt.Errorf("failed to calculate size of %s: %w", match, err))
+			continue
+		}
+
+		result.SpaceFreedMB += size
+		result.FilesRemoved++
```

Comment:
Isn't this going to count files/folders that are directly in the directory being removed, and not the total number of files/directories recursively underneath the directory being removed? Is this specified in the API?